src/idl/src/annotation.c

```diff
@@ -261,6 +286,29 @@ annotate_key(
   return IDL_RETCODE_OK;
 }
 
+static idl_retcode_t
+annotate_default(
+  idl_pstate_t *pstate,
+  idl_annotation_appl_t *annotation_appl,
+  idl_node_t *node)
+{
+  (void) node;
+  (void) annotation_appl;
+  (void) pstate;
+
+  assert(pstate);
+   /* additional requirements on default values?
+    * idl field types? members? enums? ...
+  if (annotation_appl->parameters
+   && annotation_appl->parameters->const_expr) {
+        idl_error(pstate, idl_location(annotation_appl),
+          "@default can only be assigned to values of the following types: ");
+    return IDL_RETCODE_SEMANTIC_ERROR;
+  }*/
+
+  return IDL_RETCODE_OK;
+}
+
```

Comment:
Maybe better to move the @default implementation to a different PR (including the changes below for @default in the annotations array)? Or fix the implementation in this PR (and change the PR title)